engine/server.ts

```diff
@@ -709,6 +788,71 @@ export class ChannelEngine {
     }
   }
 
+  async _handleSubtitleManifest(req, res, next) {
+    debug(`req.url=${req.url}`);
+    const session = sessions[req.params[2]];
+    if (session) {
+      try {
+        const body = await session.getCurrentSubtitleManifestAsync(
+          req.params[0],
+          req.params[1],
+          req.headers["x-playback-session-id"]
+        );
+        res.sendRaw(200, Buffer.from(body, 'utf8'), {
+          "Content-Type": "application/vnd.apple.mpegurl",
+          "Access-Control-Allow-Origin": "*",
+          "Cache-Control": `max-age=${this.streamerOpts.cacheTTL || '4'}`,
+          "X-Instance-Id": this.instanceId + `<${version}>`,
+        });
+        next();
+      } catch (err) {
+        next(this._gracefulErrorHandler(err));
+      }
+    } else {
+      const err = new errs.NotFoundError('Invalid session');
+      next(err);
+    }
+  }
+
+  async _handleDummySubtitleEndpoint(req,res,next) {
+    debug(`req.url=${req.url}`);
+    const session = sessions[req.params.channelId];
+    if (session) {
+      try {
+        const body = `WEBVTT`;
+        res.sendRaw(200, Buffer.from(body, 'utf8'), {
+          "Content-Type": "application/x-mpegURL;charset=UTF-8",
+          "Access-Control-Allow-Origin": "*",
+          "Cache-Control": `max-age=${this.streamerOpts.cacheTTL || '4'}`,
+          "X-Instance-Id": this.instanceId + `<${version}>`,
+        });
+        next();
+      } catch (err) {
+        next(this._gracefulErrorHandler(err));
+      }
+    } else {
+      const err = new errs.NotFoundError('Invalid session');
+      next(err);
+    }
+  }
+
+  async _handleSubtitleSliceEndpoint(req,res,next) {
+    debug(`req.url=${req.url}`);
+      try {
+        const slicer = new SubtitleSlicer();
+        const body = await slicer.generateVtt(req.query);
+        res.sendRaw(200, Buffer.from(body, 'utf8'), {
+          "Content-Type": "application/x-mpegURL;charset=UTF-8",
```

Comment:
use the content type `application/vnd.apple.mpegurl`

Reply:
fixed